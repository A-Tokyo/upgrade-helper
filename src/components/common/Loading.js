import React from 'react'
import styled, { keyframes } from 'styled-components'
import logo from '../../logo.svg'

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`

const Loading = styled(props => (
  <img alt="Loading" title="Loading" {...props} src={logo} />
))`
  width: 60px;
  margin: 15px 0;
  animation: ${rotate} 3s linear infinite;
`

export default Loading
