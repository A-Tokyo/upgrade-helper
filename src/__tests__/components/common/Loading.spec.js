import React from 'react'
import { render } from '@testing-library/react'
import Loading from '../../../components/common/Loading'

it('renders without crashing', () => {
  const { container } = render(<Loading logo="test.png" />)

  expect(container).toMatchSnapshot()
})
