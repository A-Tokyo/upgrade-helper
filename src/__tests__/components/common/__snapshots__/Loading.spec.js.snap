// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <img
    alt="Loading"
    class="sc-AykKC cqLuft"
    logo="test.png"
    src="loading.svg"
    title="Loading"
  />
</div>
`;
